Project-Hotpot/Project-Hotpot/Parse-Backend/PFPlaylist.swift

```diff
@@ -40,8 +40,37 @@ class PFPlaylist: PFObject, PFSubclassing {
             
         })
     }
-}
-    
-    
     
+    class func addSongToPlaylistInBackground(song: PFSong,
+                                             playlist: PFPlaylist,
+                                             completion: @escaping((Bool, Error?) -> Void)) {
+        let songID = song.objectId!
+        //TODO: consider switching from array of SongID's to relational table
+        playlist.songArray?.append(songID)
+        playlist.saveInBackground {success, error in
+            if success == true {
+                completion(true, nil)
```

Comment:
The super Swifty way of doing this is by using the `Result` enum (https://developer.apple.com/documentation/swift/result). This greatly clarifies that you can get _either_ a success _or_ an error, and keeps the handling code quite clean.

This article is a nice summary of what using `Result` looks like: https://redflowerinc.com/using-result-enum-in-swift/ 

I personally think Parse is setting a bad example here; I realize the Objective-C paradigm is to have a closure with two arguments (data, error). But when "data" is a boolean, it seems like it would be much more straightforward to just have (error), and if error is nil, the call succeeded. But they are sticking with a known convention, so I can't knock them too hard...